src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,40 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (Object.getPrototypeOf(animal) !== Carnivore.prototype
+    && animal.hidden !== true) {
```

Comment:
Use `instanceof` to check instead your construction.
https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Operators/instanceof

You can simplify the check for the hidden flag

```suggestion
    if (animal instanceof Herbivores && !animal.hidden) {
```